soh/soh/Enhancements/randomizer/3drando/logic.cpp

```diff
@@ -845,7 +847,7 @@ namespace Logic {
      ProgressiveScale      = 0;
      ProgressiveHookshot   = 0;
      ProgressiveBow        = 0;
-     ProgressiveWallet     = 0;
+     ProgressiveWallet     = ShuffleChildWallet.Is(true) ? 0 : 1;
```

Comment:
This section resets logic to the start of the playthrough, if you're not shuffling child wallet, you start with it and that has to be accounted for in the logic reset

Reply:
Alternatively we could add the child wallet in `GenerateStartingInventory()` from `starting_inventory.cpp`, but then would need to always add the extra progressive wallet to the item pool as the flow is to remove starting items from the pool, so we wouldn't want to remove a valid wallet. Ref: `RemoveStartingItemsFromPool()`.

I'm not sure what would look more clean/obvious for the intent.

Reply:
I think it's better this way because adding the child wallet to the item pool would set a precedent that would probably be followed by shuffle ocarina buttons (you have to find the ocarina buttons before being able to use them while playing songs) from n64 rando and always adding the 5 button items to then remove them would be weird.

Reply:
That's the nature of clearing vs resetting. I believe the whole point is to remove everything, whether it's default or not.

Reply:
A few lines down there's also this
```c
FireTempleKeys = IsKeysanity || Dungeon::FireTemple.IsMQ() ? 0 : 1;
```

Reply:
Yeah I think the change here is more clear, but could at least use a comment to help indicate the need (even if it might feel redundant).

Clarifying logic changes I think is very important especially as rando grows.

Reply:
Added a comment